src/routes/public.js

```diff
@@ -0,0 +1,121 @@
+import Router from 'koa-router';
+import successfullyConfirm from '../views/successfullyConfirm';
+import unsuccessfullyConfirm from '../views/unsuccessfullyConfirm';
+
+import { createOne, getByEmail } from '../models/Listeners';
+import { checkIfExists, createOne as createOneSeminarListener } from '../models/Seminars_Listeners';
+import { getAllForCurrentSeminar, getByMonth, getFirstFutureLesson } from '../models/Lessons';
+import { getAll, getOne as getOneSeminar } from '../models/Seminars';
+import { getOne as getOnePreacher } from '../models/Preachers';
+
+export const router = new Router();
+
+function getQueryParams(queryStr) {
+  const pares = queryStr.split('&');
+  const result = {};
+  pares.forEach((el) => {
+    const [key, value] = el.split('=');
+    result[key] = decodeURIComponent(value);
+  });
+  return result;
+}
+
+function getLast(arr) {
+  return arr[arr.length - 1];
+}
+
+function getPrettyDate(date) {
+  return `${date.getDate()}.${date.getMonth() + 1}.${date.getFullYear()}`;
+}
+
+router
+  .get('/preachers/:id', async (ctx) => {
+    try {
+      ctx.body = await getOnePreacher(ctx.params.id);
+    } catch (err) {
+      ctx.throw(404, 'Cannot find preacher!');
+    }
+  })
+  .get('/lessons/month/:number', async (ctx) => {
+    try {
+      const result = await getByMonth(ctx.params.number);
+      ctx.body = result.length ? result : [{ info: 'В этом месяце нет семинаров' }];
+    } catch (err) {
+      ctx.throw(404, 'Unable find lessons!');
+    }
+  })
+  .get('/seminars/current', async (ctx) => {
+    let lesson;
+    let seminarId;
+    let seminarLessons;
+    let currentSeminar;
+
+    try {
+      lesson = await getFirstFutureLesson();
+      seminarId = lesson ? lesson.seminar_id : getLast(await getAll()).id;
```

Comment:
is it possible to get last from DB? What if you have over million records in db, but you need just last one?